solver/src/driver.rs

```diff
@@ -195,10 +195,39 @@ impl Driver {
             );
             // This is an additional debug log so that the log message doesn't get too long as
             // settlement information is recoverable through tenderly anyway.
-            tracing::warn!("settlement failure for: \n{:#?}", settlement.settlement,);
+            tracing::warn!("settlement failure for: \n{:#?}", settlement,);
         }
     }
 
+    // Rate settlements, ignoring those for which the rating procedure failed.
+    async fn rate_settlements(
+        &self,
+        settlements: Vec<Settlement>,
+        prices: &HashMap<H160, BigRational>,
+    ) -> Vec<RatedSettlement> {
+        use futures::stream::StreamExt;
+        futures::stream::iter(settlements)
+            .filter_map(|settlement| async {
+                let surplus = settlement.total_surplus(prices);
+                if let Ok(gas_estimate) = settlement_submission::estimate_gas(
+                    &self.settlement_contract,
+                    &settlement.clone().into(),
+                )
+                .await
```

Comment:
nit: Instead of if/else you might be able to get away with ?

```rust
let gas_estimate = ettlement_submission::estimate_gas(
                    &self.settlement_contract,
                    &settlement.clone().into(),
                )
                .await?
Ok(RatedSettlement {
                        settlement,
                        surplus,
                        gas_estimate,
                    })

```